### PR TITLE
#345: Weekend ticket prices

### PR DESCRIPTION
**Pull request by lukybil**
- display weekend and holidays price

### CODE REVIEW

File: app/(app)/zone-details.tsx [hunk 1]
```diff
@@ -28,13 +29,18 @@ const ZoneDetailsScreen = () => {
     return null
   }
 
+  const price = getPriceFromZone(zone)
+
+  // eslint-disable-next-line eqeqeq
+  const isAllWeekPrice = zone.weekendsAndHolidaysPrice == undefined
```
Comment (PavelFilo) [inline app/(app)/zone-details.tsx:35]: why only `==`?

File: utils/getPriceFromZone.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+const holidays = new Holidays('SK')
+
+export const getPriceFromZone = (
+  zone: MapUdrZone | MapUdrZoneWithTranslationProps,
+  date?: Date,
+) => {
+  const localDate = date ?? new Date()
+  // eslint-disable-next-line eqeqeq
+  if (zone.weekendsAndHolidaysPrice == undefined) {
```
Comment (PavelFilo) [inline utils/getPriceFromZone.ts:13]: same here, why?

File: app/(app)/zone-details.tsx [hunk 3]
```diff
@@ -57,11 +63,20 @@ const ZoneDetailsScreen = () => {
           </FlexRow>
         ) : null}
         {/* eslint-disable-next-line eqeqeq */}
+        {isAllWeekPrice || zone.price == undefined ? null : (
```
Comment (PavelFilo) [inline app/(app)/zone-details.tsx:66]: also here... if you want to check if its null or undefined create a function or add there both undefined and null with ===

**Review by PavelFilo — APPROVED** [review]

File: app/(app)/zone-details.tsx (hunk 3 above)
Comment (lukybil) [reply app/(app)/zone-details.tsx:66]: changed to `== null` since this is also allowed by eslint

Comment (radoslavzeman) [reply app/(app)/zone-details.tsx]: Last time we discussed this in the office, we agreed that `== null` is hardly readable a prone to bugs, so it's better to use `isDefined` function, that we have in almost all projects.
This is just a note for future :)

**Pull request merged**